tests/test_dispatch.py

```
@@ -40,6 +40,17 @@ def test_dispatch_invalid_tier() -> None:
         dispatch(action="understand", provider="gemini", tier="mega")
 
 
+def test_dispatch_invalid_image_url_scheme() -> None:
+    with pytest.raises(ValueError, match="Invalid image_url scheme: 'file'. Must be 'http' or 'https'"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)